### PR TITLE
Persistence classes

### PR DESCRIPTION
**Pull request by nicodh**
Adaptions to this breaking change: https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/10.0/Breaking-87623-ReplaceConfigpersistenceclassesTyposcriptConfiguration.html

### CODE REVIEW

File: Classes/Configuration/ExtensionBuilderConfigurationManager.php [hunk 1]
```diff
@@ -155,10 +155,10 @@ public function getExtensionBuilderSettings(): array
      * @param string $extensionKey
      * @return array settings
      */
-    public function getExtensionSettings(string $extensionKey): array
+    public function getExtensionSettings(string $extensionKey, string $extensionStoragePath): array
```
Comment (simonschaufi) [inline Classes/Configuration/ExtensionBuilderConfigurationManager.php:158]: docheader is missing the new parameter

File: Classes/Configuration/ExtensionBuilderConfigurationManager.php [hunk 2]
```diff
@@ -228,9 +228,9 @@ public function getRecordType($className)
      * @param string $extensionKey
      * @return string path
      */
-    public function getSettingsFile($extensionKey): string
+    public function getSettingsFile($extensionKey, $storagePath): string
```
Comment (simonschaufi) [inline Classes/Configuration/ExtensionBuilderConfigurationManager.php:231]: docheader is missing the new parameter and it should be defined as string

File: Classes/Domain/Model/DomainObject.php [hunk 3]
```diff
@@ -497,17 +497,26 @@ public function getHasBooleanProperties()
     /**
      * @return array
      */
-    public function hasPropertiesThatNeedMappingStatements()
+    public function getHasPropertiesWithMappingStatements()
```
Comment (simonschaufi) [inline Classes/Domain/Model/DomainObject.php:500]: return is bool and not array

File: Classes/Domain/Model/DomainObject.php [hunk 4]
```diff
@@ -497,17 +497,26 @@ public function getHasBooleanProperties()
...
-            }
-        }
-        return $propertiesWithMappingStatements;
+        $propertiesWithMappingStatements = $this->getPropertiesThatNeedMappingStatements();
+        return count($propertiesWithMappingStatements) > 0;
     }
 
+    /**
+    * @return array
+    */
```
Comment (simonschaufi) [inline Classes/Domain/Model/DomainObject.php:508]: indentation wrong

File: Classes/Domain/Model/DomainObject/AbstractProperty.php [hunk 5]
```diff
@@ -472,7 +472,8 @@ public function getDomainObject(): DomainObject
     public function getMappingStatement(): ?string
     {
         if ($this->getFieldName() != GeneralUtility::camelCaseToLowerCaseUnderscored($this->name)) {
-            return $this->getFieldName() . '.mapOnProperty = ' . $this->name;
+            $statement = "'" . $this->name . "' => [\n\t'fieldName' => '". $this->getFieldName() . "'\n],";
```
Comment (simonschaufi) [inline Classes/Domain/Model/DomainObject/AbstractProperty.php:475]: please make this a oneliner. assigning a variable is not needed

File: Tests/Fixtures/TestExtensions/test_extension/ExtensionBuilder.json [hunk 6]
```diff
@@ -534,6 +534,7 @@
             }
         }
     ],
+	"storagePath": "\/var\/www\/html\/public\/typo3conf\/ext\/",
```
Comment (simonschaufi) [inline Tests/Fixtures/TestExtensions/test_extension/ExtensionBuilder.json:537]: indentation looks wrong

**Review by simonschaufi — CHANGES_REQUESTED** [review]

File: Classes/Domain/Model/DomainObject.php [hunk 7]
```diff
@@ -495,19 +495,27 @@ public function getHasBooleanProperties()
     }
 
     /**
-     * @return array
+     * @return boolean
```
Comment (liayn) [inline Classes/Domain/Model/DomainObject.php:498]: it is `bool`

File: Classes/Service/FileGenerator.php [hunk 8]
```diff
@@ -917,6 +920,23 @@ public function generateControllerTests($controllerName, DomainObject $domainObj
         ]);
     }
 
+    public function generateExtbaseConfigClass() {
```
Comment (simonschaufi) [inline Classes/Service/FileGenerator.php:923]: this is not PSR-2. the `{` should be on the next line

File: Classes/Service/FileGenerator.php [hunk 9]
```diff
@@ -917,6 +920,23 @@ public function generateControllerTests($controllerName, DomainObject $domainObj
...
+               }
+               $fileContents = $this->renderTemplate('Configuration/Extbase/Persistence/Classes.phpt', [
+                   'extension' => $this->extension
+               ]);
+               $this->writeFile($this->configurationDirectory . '/Extbase/Persistence/Classes.php', $fileContents);
+           }
+        } catch (\Exception $e) {
+           throw new \Exception('Could not generate Extbase Persistence Class Configuration, error: ' . $e->getMessage());
+        }
+
```
Comment (simonschaufi) [inline Classes/Service/FileGenerator.php:937]: remove empty line

File: Classes/Service/FileGenerator.php [hunk 10]
```diff
@@ -520,16 +522,17 @@ protected function generateTyposcriptFiles()
...
+                if ($this->extension->getDomainObjectsThatNeedMappingStatements()) {
+                    $fileContents = $this->generateStaticTyposcript();
+                    $this->writeFile($this->extensionDirectory . 'ext_typoscript_setup.typoscript', $fileContents);
+                }
+            } catch (\Exception $e) {
+                throw new \Exception('Could not generate static typoscript, error: ' . $e->getMessage());
             }
-        } catch (\Exception $e) {
-            throw new \Exception('Could not generate static typoscript, error: ' . $e->getMessage());
+
```
Comment (simonschaufi) [inline Classes/Service/FileGenerator.php:535]: remove empty line

File: Classes/Configuration/ExtensionBuilderConfigurationManager.php (hunk 2 above)
Comment (simonschaufi) [reply Classes/Configuration/ExtensionBuilderConfigurationManager.php:230]: please define the parameter type also directly in the method

**Review by simonschaufi — CHANGES_REQUESTED** [review]

**Pull request merged**